src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -5619,14 +5621,17 @@ private Rope executeConcat(Rope left, Rope right, RubyEncoding compatibleEncodin
             return concatNode.executeConcat(left, right, compatibleEncoding.jcoding);
         }
 
-        private RubyEncoding executeCheckEncoding(Object string, Object other) {
+        private RubyEncoding executeCheckEncoding(RopeWithEncoding string, RopeWithEncoding other) {
```

Comment:
Because there is a single specialization and this is always used it's better to used `@Cached` for the CheckRopeEncodingNode. Was already the wrong though.